packages/foundation/src/createComponent.tsx

```diff
@@ -63,15 +75,69 @@ export function createComponent<
     }
 
     const theme = componentProps.theme || settings.theme;
+    const tokens = _resolveTokens(componentProps, theme, options.tokens, settings.tokens, componentProps.tokens) as any;
+    let styles;
+
+    const finalStyles: { [key: string]: string | undefined } = {};
+
+    if (!options.disableCaching) {
+      // We get the entry in the memoized classNamesMap for the current component or create one if it doesn't exist.
+      const displayName = options.displayName || view.name;
+      if (!memoizedClassNamesMap.hasOwnProperty(displayName)) {
```

Comment:
We probably shouldn't fall back to `view.name` here because it will be minified nonsense in production. Similar to the use in `_getCustomizations` above, it should just rely on `options.displayName`.

What would happen if Fabric or an end user happened to have a name collision with an existing component? What would be the side effects of that name collision here?

Reply:
But `displayName` can be `undefined`, where do we map in that case? Do we just not cache?

Reply:
Falling back to not caching seems reasonable. Customizations would fail as well. `displayName` is essentially functionally required, just not in typing. It's something we could consider requiring, but we also wanted the entire `options` arg to be optional for ease of use.